packages/contracts/src/healthcheck/HealthChecker.sol

```diff
@@ -86,17 +100,76 @@ abstract contract HealthChecker is SafeInitializable, OwnableUpgradeable {
             return;
         }
 
-        // Perform the health check, revert the transaction if unsuccessful.
-        bool success = healthCheck.check(
-            strategy,
-            profit,
-            loss,
-            debtPayment,
-            debtOutstanding,
-            totalDebt
-        );
-        if (!success) {
+        // If no custom health check implementation provided, call default one.
+        uint8 checkResult = 0;
+        if (address(healthCheck) == address(0)) {
+            checkResult = _defaultCheck(
+                strategy,
+                profit,
+                loss,
+                totalDebt
+            );
+        } else {
+            // Perform the health check, to revert the transaction
+            checkResult = healthCheck.check(
+                strategy,
+                profit,
+                loss,
+                debtPayment,
+                debtOutstanding,
+                totalDebt,
+                gasCost
+            );
+        }
+
+        if(checkResult == 2) { // call fallback if loss to big
+            healthCheckFallback();
+        } else if(checkResult == 1) { // call secondary fallback if loss is acceptable
```

Comment:
This 2 and 1 checks very easy to miss. Use constants instead of it
```suggestion
        if(checkResult == SHUTDOWN_STRATEGY) { // call fallback if loss to big
            healthCheckFallback();
        } else if(checkResult == STOP_TRANSACTION) { // call secondary fallback if loss is acceptable
```

Reply:
added constants